### PR TITLE
feat(versioning): add dependency versioning utility and docs #11

### PR DESCRIPTION
**Pull request by jason-capsule42**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Fixes:** #11

## Summary:

Added two new utility scripts and documentation. This feature adds all JS needed to dynamically name component dependencies based on the installed version.

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [ ] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

**Comment by blackfalcon** [comment]
@jason-capsule42 you mention that this work is being moved from this PR https://github.com/AlaskaAirlines/auro-combobox/pull/148 to this repo now as a remote resource. Is there a PR coming then I assume that will remove that code from the combobox element and then update to point to this resource? 

If that is the case, it would be nice to have that described in an issue/PR so that it's clearly documented. 

Also, https://github.com/AlaskaAirlines/auro-library/issues/11 is listed as `in progress`, is this PR ready for review?

File: docs/dependencyTagVersioning.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+3. The `build:version` script in `package.json` should be added as the first step of the `build` script.
+
+```json
+"build": "npm-run-all build:version ... etc.",
+```
+
+#### Execution
+
+Once configuration is complete, exeucte `npm run build`. This must be done once before `npm run dev` when developing locally. When Auro dependencies are initially installed or updated to new versions then `npm run build:version` or a complete `npm run build` must be executed.
```
Comment (blackfalcon) [inline docs/dependencyTagVersioning.md:33]:
```suggestion
Once the configuration is complete, execute `npm run build`. This must be done once before `npm run dev` when developing locally. When Auro dependencies are initially installed or updated to new versions then `npm run build:version` or a complete `npm run build` must be executed.
```

**Review by blackfalcon — DISMISSED** [review]

**Comment by jason-capsule42** [comment]
> @jason-capsule42 you mention that this work is being moved from this PR [AlaskaAirlines/auro-combobox#148](https://github.com/AlaskaAirlines/auro-combobox/pull/148) to this repo now as a remote resource. Is there a PR coming then I assume that will remove that code from the combobox element and then update to point to this resource?

The ticket and PR for combobox that you linked is still in progress. It will get a final commit to use the published version of these resource files from Auro Library after we release it here.

File: docs/dependencyTagVersioning.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+```js
+import { AuroDependencyVersioning } from "../scripts/dependencyTagVersioning.mjs";
+import { AuroDropdown } from '@aurodesignsystem/auro-dropdown/src/auro-dropdown.js';
+import dropdownVersion from './dropdownVersion';
+```
+
+In the components constructor add the following:
+
+```js
+this.dropdownTag = versioning.generateTag('auro-dropdown', dropdownVersion, AuroDropdown);
```
Comment (Izook) [inline docs/dependencyTagVersioning.md:65]: Should we include [this line](https://github.com/AlaskaAirlines/auro-combobox/pull/148/files#diff-94513de8f740c27b4429eb6e2f05ab7dfa54fbef9274015d546eef3e4c420f78R64) above this to make it clear that `versioning` comes from `AuroDependencyVersioning`?

**Review by Izook — APPROVED** [review]
Yugi Thumb

**Pull request merged**